### DISCUSSION TITLE
Keep ASSA styles in text box

### DISCUSSION THREAD

**Discussion by niksedk** [SubtitleEdit/subtitleedit#5098] [Ideas]
I'm thinking about keeping ASSA style italic/fon tags in the text box / list view.
Probably better for people who work a lot with ASSA...

![image](https://user-images.githubusercontent.com/1261331/120844834-45069380-c570-11eb-95d9-9c08febd6d89.png)

https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.1/SubtitleEditBeta.zip

Comments or ideas?

**Comment by OmrSi** [comment]
You mean instead of converting them to HTML ones?
I'm 100% with that. I have to replace `\` with `\\` each time I work with an ASS file because SE converts them to HTML tags otherwise.

I think I opened an issue about this 2 years ago, you told me then that this is just how SE works.

**Reply by niksedk** [reply]
> 
> 
> You mean instead of converting them to HTML ones?
> I'm 100% with that. I have to replace `\` with `\\` each time I work with an ASS file because SE converts them to HTML tags otherwise.
> 
> I think I opened an issue about this 2 years ago, you told me then that this is just how SE works.

Cool :)
Your issue post is here: https://github.com/SubtitleEdit/subtitleedit/issues/3046
SE's support of ASSA has improved a lot in the last versions, and for a bit more advanced stuff (not just italic/color tags) the mixing of html and assa tags is rather annoying...

**Reply by OmrSi** [reply]
Agreed.

**Reply by niksedk** [reply]
Committed to master and beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.1/SubtitleEditBeta.zip

**Comment by OmrSi** [comment]
@LeonCheung I think this is of interest to you as well.

**Reply by LeonCheung** [reply]
Definitely! Just.. I don't quite follow what this feature means... Is there any difference from the latest official release? 😂

**Reply by OmrSi** [reply]
SE used to convert Ass tags to HTML ones when opening an Ass file.
So if you have a line: `{\b1}Hey.` SE used to make it `<b>Hey.</b>` which was annoying in my opinion because it made it hard to deal with other tags.

Now, SE should leave the Ass tags as they are.

**Reply by LeonCheung** [reply]
Thanks for the explanation! Then I'm 100% with it too.

**Comment by OmrSi** [comment]
@niksedk That shortcut is pretty cool!

Although, can you make it work even if the file isn't ASS?
Sometimes we would just add ASS tags in an Srt file, which would work on a player just like the `{\an}` tags.

**Reply by OmrSi** [reply]
Just to make a point, here is a random Srt sub, check the first line: https://subscene.com/subtitles/the-conjuring-the-devil-made-me-do-it/arabic/2493381

**Comment by OmrSi** [comment]
Issues I'm seeing so far:
1. The color tags are `{\c&H<bbggrr>&}` and not `{\c&H<bbggrr>&}`, the should be an `&` after the color just like the alpha codes.
2. When trying to use the shortcut at the end of the tag before the closing `}`, it doesn't work.
3. When using the shortcut just after the opening `{` and adding a tag, it adds its own curlies and becomes `{\a3&H<aa>&}\2c&HFFFFFF}`
4. When adding two tags in the same line near each other, they should use one set of curlies, not more.

**Reply by OmrSi** [reply]
As an idea of how to fix 3 and 4, you can replace `}}` with `}` and `{{` with `{` and `}{` with nothing after inserting a tag.

**Comment by mammadno** [comment]
Is there any way to paste the text into SE and preserve its formatting?
For example, let's say you have a Word document with all your video scripts written out, complete with different colors for emphasis. Is there a way to transfer that text to SE without losing the colors?

![Color2](https://github.com/user-attachments/assets/76f64933-c9e3-447e-9be7-34a3777362f6)
